### PR TITLE
Fixes to support 802.15.4 on nRF54H20

### PR DESCRIPTION
**Pull request by piotrkoziar**
Content:
1. cherry-picks https://github.com/zephyrproject-rtos/zephyr/pull/70222
2. other fixes - see https://github.com/zephyrproject-rtos/zephyr/pull/73156

### CODE REVIEW

**Comment by jciupis** [comment]
For all commits cherry-picked directly from Zephyr main, please do: `git cherry-pick -x <sha>` and then prefix the commit title with `[nrf fromtree]` sauce tag.

For all the remaining commits:
1. Fix commit messages so that every one has a title, body and sign-off and does not exceed maximum line length allowed in Zephyr.
2. Open PR(s) to Zephyr with those commits and cherry-pick them with `[nrf fromlist]` sauce tag and link to upstream PR in the cherry-picked commit message body. Alternatively, add `[nrf noup]` sauce tag to commits that were agreed to only go downstream.

Also, note that commits cherry-picked from https://github.com/zephyrproject-rtos/zephyr/pull/70222 might conflict with https://github.com/nrfconnect/sdk-zephyr/pull/1721 since they are pulled directly from upstream there.

File: modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_clock_zephyr.c
Comment (jciupis) [inline modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_clock_zephyr.c:1]: Personally, I think it would be cleaner to structure the code like this:
```c
#if defined(CONFIG_CLOCK_CONTROL_NRF)
/* The entire, unmodified implementation using clock_control APIs */
#elif NRF54H_SERIES
/* Only nrf_802154_clock_hfclk_start and nrf_802154_clock_hfclk_stop
 * implemented without any additional static functions */
#else
#error "Some descriptive error message"
#endif
```

But I'm not going to block the PR over this, just a suggestion. It would be cleaner to later remove the changes once clock control is implemented for nRF54H.

File: subsys/ipc/ipc_service/backends/ipc_icbmsg.c
Comment (jciupis) [inline subsys/ipc/ipc_service/backends/ipc_icbmsg.c:544]: Isn't it the number of sent _blocks_?

File: modules/hal_nordic/Kconfig
Comment (jciupis) [inline modules/hal_nordic/Kconfig:58]: These two lines could be expressed as:
```
default y if !SOC_NRF54H20
```

File: drivers/ieee802154/ieee802154_nrf5.c
Comment (jciupis) [inline drivers/ieee802154/ieee802154_nrf5.c:132]: Please provide this implementation as an nRF54H20-only option, not replace the existing implementation that seemingly worked well for other devices so far.

File: modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_clock_zephyr.c
Comment (piotrkoziar) [reply modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_clock_zephyr.c:1]: I applied changes inspired by this. Good idea with the #error

File: drivers/ieee802154/ieee802154_nrf5.c
Comment (ankuns) [inline drivers/ieee802154/ieee802154_nrf5.c:135]: I think that we should provide the workaround code only when `NRF54H_SERIES` is defined while keeping old behavior while it is not defined, so:
```c
#if defined(NRF54H_SERIES)
	/* Can't access SICR with device id on a radio core. Use BLE.ADDR. */
	deviceid[0] = NRF_FICR->BLE.ADDR[0];
	deviceid[1] = NRF_FICR->BLE.ADDR[1];
#elif defined(CONFIG_TRUSTED_EXECUTION_NONSECURE) && defined(NRF_FICR_S)
	soc_secure_read_deviceid(deviceid);
#else
	deviceid[0] = nrf_ficr_deviceid_get(NRF_FICR, 0);
	deviceid[1] = nrf_ficr_deviceid_get(NRF_FICR, 1);
#endif
```

Comment (piotrkoziar) [reply drivers/ieee802154/ieee802154_nrf5.c]: Removed the dependency to hwinfo and applied the suggestion.

**Review by ankuns — APPROVED** [review]

**Comment by piotrkoziar** [comment]
The changes were pulled with the recent upmerge.

**Pull request closed without merging**
